### ISSUE TITLE
[Adventure] Scoreboard Team Prefix Empty Crashes Player

### ISSUE THREAD

**Issue by hawkfalcon** [PaperMC/Paper#5371]
### Link/paste of stack trace
On Death
```[09:40:09 WARN]: Failed to handle packet for /127.0.0.1:60600
net.minecraft.server.v1_16_R3.ReportedException: Ticking player
	at net.minecraft.server.v1_16_R3.EntityPlayer.playerTick(EntityPlayer.java:601) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.PlayerConnection.tick(PlayerConnection.java:178) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.NetworkManager.a(NetworkManager.java:390) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.ServerConnection.c(ServerConnection.java:145) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1389) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:379) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1216) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1004) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.5.jar:git-Paper-542]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.IllegalArgumentException: Don't know how to serialize io.papermc.paper.adventure.AdventureComponent@65873512 as a Component
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:323) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:277) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:298) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:105) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:81) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.Gson.toJson(Gson.java:669) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.Gson.toJsonTree(Gson.java:562) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.Gson.toJsonTree(Gson.java:541) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.b(IChatBaseComponent.java:359) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.toJsonTree(IChatBaseComponent.java:357) ~[patched_1.16.5.jar:git-Paper-542]
	at io.papermc.paper.adventure.PaperAdventure.asAdventure(PaperAdventure.java:124) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityPlayer.die(EntityPlayer.java:678) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityLiving.damageEntity(EntityLiving.java:1268) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityHuman.damageEntity(EntityHuman.java:761) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityPlayer.damageEntity(EntityPlayer.java:842) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityLiving.an(EntityLiving.java:2016) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.Entity.doVoidDamage(Entity.java:573) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.Entity.performVoidDamage(Entity.java:568) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.Entity.entityBaseTick(Entity.java:469) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityLiving.entityBaseTick(EntityLiving.java:266) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.Entity.tick(Entity.java:406) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityLiving.tick(EntityLiving.java:2449) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityHuman.tick(EntityHuman.java:156) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.EntityPlayer.playerTick(EntityPlayer.java:518) ~[patched_1.16.5.jar:git-Paper-542]
	... 9 more
```
On subsequent join
```[21:48:01 INFO]: Hawkfalcon lost connection: Internal server error
[21:48:10 WARN]: Failed to handle packet for /127.0.0.1:59742
java.lang.IllegalArgumentException: Don't know how to serialize io.papermc.paper.adventure.AdventureComponent@356c51b9 as a Component
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:323) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:277) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:298) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.serialize(IChatBaseComponent.java:105) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:81) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.Gson.toJson(Gson.java:669) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.Gson.toJsonTree(Gson.java:562) ~[patched_1.16.5.jar:git-Paper-542]
	at com.google.gson.Gson.toJsonTree(Gson.java:541) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.b(IChatBaseComponent.java:359) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.IChatBaseComponent$ChatSerializer.toJsonTree(IChatBaseComponent.java:357) ~[patched_1.16.5.jar:git-Paper-542]
	at io.papermc.paper.adventure.PaperAdventure.asAdventure(PaperAdventure.java:124) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.PlayerList.postChunkLoadJoin(PlayerList.java:270) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.PlayerList.lambda$null$1(PlayerList.java:218) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.PlayerConnection.tick(PlayerConnection.java:169) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.NetworkManager.a(NetworkManager.java:390) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.ServerConnection.c(ServerConnection.java:145) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1389) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:379) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1216) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1004) ~[patched_1.16.5.jar:git-Paper-542]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.5.jar:git-Paper-542]
	at java.lang.Thread.run(Thread.java:832) [?:?]```

### Plugin list:
Just a custom one, which uses adventure, and Vault. I do use this library https://github.com/MrMicky-FR/FastBoard

### Actions to reproduce (if known):
For a scoreboard Team:
team.prefix(Component.empty().color(textColor));
I also tried Component.text().color(textColor).asComponent());

This doesn't break right away. 
When the player dies, they get kicked with 'Internal Server Error'. 
Any subsequent rejoin kicks with same message (I do add the player to the team on join, though)

### Paper version:
___Paste the output of running `/version` on your server WITH the Minecraft version. `latest` is not a version; we require the output of `/version` so we can properly track down the issue.___
git-Paper-551 (MC: 1.16.5)

**Comment by hawkfalcon** [comment]
I confirmed it happens without the Scoreboard lib I have is not the culprit; it still occurs when removing it completely

**Comment by MiniDigger** [comment]
(for reference, code that causes this [scoreboard]team.prefix(Component.empty().color(textColor));)

**Comment by hawkfalcon** [comment]
I did some digging and was able to reliably reproduce it on it's own
```
    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        Team team = player.getScoreboard().getTeam(player.getName());
        if (team != null) {
            team.color(NamedTextColor.BLUE);
            team.prefix(Component.text("ASDF").color(NamedTextColor.BLUE).asComponent());
        }
    }
```
If ALL I do it set color, it works.

**Comment by jpenilla** [comment]
fixed in 0aea6c2ff31ebe726b567e1e4a0b3b701995f05d
